mesonbuild/mesonlib.py

```diff
@@ -1332,57 +1409,63 @@ def detect_subprojects(spdir_name, current_dir='', result=None):
                 result[basename] = [trial]
     return result
 
+
 def substring_is_in_list(substr: str, strlist: T.List[str]) -> bool:
     for s in strlist:
         if substr in s:
             return True
     return False
 
-class OrderedSet(collections.abc.MutableSet):
+
+class OrderedSet(T.MutableSet[_T]):
     """A set that preserves the order in which items are added, by first
     insertion.
     """
-    def __init__(self, iterable=None):
-        self.__container = collections.OrderedDict()
+    def __init__(self, iterable: T.Optional[T.Iterable[_T]] = None):
```

Comment:
```suggestion
    def __init__(self, iterable: T.Optional[T.Iterable[_T]] = None) -> None:
```

Reply:
We seem not not use a return type on `__init__` when it has arguments, so for now I'm going to leave this as-is, I'd rather do a batch '-> None:` to all `__init__`s if we're going to do that.